bulbs/content/tasks.py

```diff
@@ -51,3 +52,93 @@ def update_feature_type_rates(featuretype_pk):
                 rate=0,
                 feature_type_id=featuretype_pk,
                 role_id=role.pk)
+
+
+@shared_task(default_retry_delay=5)
+def post_to_instant_articles_api(content_pk):
+    import requests
+    import logging
+
+    from django.conf import settings
+    from django.template.loader import render_to_string
+    from django.template.base import TemplateDoesNotExist
+
+    from bulbs.utils import vault
+    from bulbs.instant_articles.renderer import InstantArticleRenderer
+    from bulbs.instant_articles.transform import transform
+    from .models import Content
+
+    if getattr(settings, 'FACEBOOK_API_ENV', '').lower() == 'production':
+        logger = logging.getLogger(__name__)
+
+        content = Content.objects.get(pk=content_pk)
+        feature_type = getattr(content, 'feature_type', None)
+
+        fb_page_id = getattr(settings, 'FACEBOOK_PAGE_ID', '')
+        fb_api_url = getattr(settings, 'FACEBOOK_API_BASE_URL', '')
+
+        fb_access_token = vault.read(settings.FACEBOOK_TOKEN_VAULT_PATH)
+
+        if feature_type and feature_type.instant_article and content.is_published:
+            context = {
+                "content": content,
+                "absolute_uri": getattr(settings, 'WWW_URL'),
+                "transformed_body": transform(
+                    getattr(content, 'body', ""),
+                    InstantArticleRenderer())
+            }
+            try:
+                source = render_to_string(
+                    "instant_article/_instant_article.html", context
+                )
+            except TemplateDoesNotExist:
+                source = render_to_string(
+                    "instant_article/base_instant_article.html", context
+                )
+
+            # Post article to instant article API
+            post = requests.post(
+                '{0}/{1}/instant_articles'.format(fb_api_url, fb_page_id),
+                data={
+                    'access_token': fb_access_token,
+                    'html_source': source,
+                    'published': 'true',
+                    'development_mode': 'false'
+                })
+
+            # Get status of article
+            status = requests.get('{0}/{1}?access_token={2}'.format(
+                fb_api_url,
+                post.json()['id'],
+                fb_access_token
+            ))
+
+            # If upload if successful, set ia_id to id
+            if status.json()['status'] == 'SUCCESS':
+                # using a queryset.update() will stop save() from recursing
+                Content.objects.filter(pk=content_pk).update(instant_article_id=status.json()['id'])
+            else:
+                logger.error('Error in posting to Instant Article API: {}'.format(
+                    status.json()
+                ))
+        # if article is being unpublished, delete it from IA API
+        elif (feature_type and
+              feature_type.instant_article and
```

Comment:
Do you need these first 2 "feature type" conditions? Isn't it enough that the content has an "instant_article_id"? 

One bad situation would be if a particular feature type (say NIB) turned off IA, and then future NIB article deletes wouldn't remove the legacy IA content.